src/main/java/com/google/step/YOUR_PROJECT_NAME_HERE/servlets/SearchServlet.java

```diff
@@ -15,23 +15,65 @@
 package com.google.step.YOUR_PROJECT_NAME_HERE.servlets;
 
 import java.io.IOException;
-import java.util.Arrays;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import javax.servlet.ServletException;
+import org.json.JSONObject;
 
 /** Servlet that handles searches. */
 @WebServlet("/search")
 public class SearchServlet extends HttpServlet {
 
+  private final HttpClient httpClient =
+      HttpClient.newBuilder().version(HttpClient.Version.HTTP_2).build();
+
+  private static String encodeValue(String value) {
+    try {
+      return URLEncoder.encode(value, StandardCharsets.UTF_8.toString());
+    } catch (UnsupportedEncodingException ex) {
+      throw new RuntimeException(ex.getCause());
+    }
+  }
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
-    // Set some data on the request for the template to access.
-    request.setAttribute("previousQuery", request.getParameter("q"));
-    request.setAttribute("results", Arrays.asList("Aaron", "Iris", "Kayla"));
-    // Forward the request to the template (which is a servlet itself).
-    request.getRequestDispatcher("WEB-INF/templates/search.jsp").forward(request, response);
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
+    String query = encodeValue(request.getParameter("q"));
```

Comment:
I'm thinking in order to avoid future repetition, this could be written in a separate method (w/ CSE and query as parameters, returning the link) since we're doing this for StackOverflow & GeeksForGeeks as well.